libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -34,6 +34,12 @@ internal class RouteAlternativesController(
     private val observers = CopyOnWriteArraySet<RouteAlternativesObserver>()
     private var currentRequestId: Long? = null
 
+    fun triggerAlternativeRequest() {
+        currentRequestId?.let { directionsSession.cancelRouteRequest(it) }
+        restartTimer()
+        requestRouteAlternatives()
+    }
```

Comment:
```suggestion
    fun triggerAlternativeRequest() {
        interrupt()
        requestRouteAlternatives()
    }
```

we need to cancel ongoing request and restart timer only if there is at least one observer. `interrupt()` does exactly the same

Reply:
> only if there is at least one observer.

Good call